cargo-near/src/commands/build/actions/non_reproducible_wasm/mod.rs

```diff
@@ -0,0 +1,213 @@
+            for pair in self.env.iter() {
+                pair.split_once('=').ok_or(color_eyre::eyre::eyre!(
+                    "invalid \"key=value\" environment argument (must contain '='): {}",
+                    pair
+                ))?;
+            }
+            Ok(())
+        }
+    }
+
+    pub(super) fn get_key_vals(input: Vec<String>) -> Vec<(String, String)> {
+        let iterator = input.iter().flat_map(|pair_string| {
+            pair_string
+                .split_once('=')
+                .map(|(env_key, value)| (env_key.to_string(), value.to_string()))
+        });
+
+        let dedup_map: HashMap<String, String> = HashMap::from_iter(iterator);
+
+        let result = dedup_map.into_iter().collect();
+        tracing::info!(
+            target: "near_teach_me",
+            parent: &tracing::Span::none(),
+            "Passed additional environment pairs:\n{}",
+            near_cli_rs::common::indent_payload(&format!("{:#?}", result))
+        );
+        result
+    }
+}
+
+pub mod rule {
+    use color_eyre::Section;
+    use colored::Colorize;
+
+    const COMMAND_ERR_MSG: &str = "`container_build_command` is required to start with";
+
+    fn is_inside_docker_context() -> bool {
+        std::env::var(cargo_near_build::env_keys::nep330::BUILD_ENVIRONMENT).is_ok()
+    }
+    pub fn assert_locked(opts: &super::BuildOpts) {
+        if is_inside_docker_context() {
+            assert!(
+                opts.locked,
+                "build command should have `--locked` flag in docker"
+            );
+        }
+    }
+
+    fn get_docker_image() -> String {
+        std::env::var(cargo_near_build::env_keys::nep330::BUILD_ENVIRONMENT).expect(&format!(
+            "`{}` is set",
+            cargo_near_build::env_keys::nep330::BUILD_ENVIRONMENT
+        ))
+    }
+    pub fn enforce_this_program_args() -> color_eyre::eyre::Result<()> {
+        if is_inside_docker_context() {
+            let args = std::env::args().collect::<Vec<_>>();
+            let default_cmd =
+                cargo_near_build::BuildOpts::default().get_cli_command_for_lib_context();
+            let default_cmd_len = default_cmd.len();
+            if (args.len() < default_cmd_len)
+                || (args[1..default_cmd_len] != default_cmd[1..default_cmd_len])
+            {
+                return Err(color_eyre::eyre::eyre!(
+                    "{}\n`{}` for the used image:\n{}",
+                    COMMAND_ERR_MSG,
+                    serde_json::to_string(&default_cmd).unwrap(),
+                    get_docker_image()
+                )
+                .note(format!(
+                    "The default `{}` has changed since `{}` image\n\
+                    See {}",
+                    "container_build_command".cyan(),
+                    "sourcescan/cargo-near:0.13.0-rust-1.83.0".cyan(),
+                    "https://github.com/near/cargo-near/releases".cyan()
+                )));
```

Comment:
this would look like the following, if someone updates `image` and `image_digest`, 
but doesn't update `container_build_command`  (which stays to be `["cargo", "near", "build"]`):

![image](https://github.com/user-attachments/assets/c03f4b1f-e9c4-4e36-b074-bd08a220f11b)